account_credit_control/models/account_invoice.py

```diff
@@ -30,6 +30,88 @@ class AccountInvoice(models.Model):
         readonly=True,
         copy=False,
     )
+    credit_control_notes = fields.Char(
+        compute='_compute_credit_control_notes',
+        inverse='_inverse_credit_control_notes',
+        readonly=True,
+        states={'open': [('readonly', False)]}
+    )
+    credit_control_date = fields.Date(
+        compute='_compute_credit_control_date',
+        inverse='_inverse_credit_control_date',
+        string='Credit Control Ignore Before',
+        readonly=True,
+        states={'open': [('readonly', False)]}
+    )
+
+    @api.multi
+    def _inverse_credit_control_notes(self):
+        """
+        Set credit control notes on every move line if invoice notes
+        have been modified
+        :return:
+        """
+        invoices = self.filtered('move_id')
+        # Search all moves once
+        moves = invoices._get_all_related_move_line()
+        for invoice in invoices:
+            invoice_moves = moves.filtered(
+                lambda m: m.invoice_id == invoice.id)
+
+            for invoice_move in invoice_moves:
+                invoice_move.credit_control_notes =\
+                    invoice.credit_control_notes
+                break
+
+    @api.multi
+    def _compute_credit_control_notes(self):
+        invoices = self.filtered('move_id')
+        # Search all moves once
+        moves = invoices._get_all_related_move_line()
+        for invoice in invoices:
+            invoice_moves = moves.filtered(
+                lambda m: m.invoice_id.id == invoice.id).with_context(
+                from_parent_object=True)
+            for invoice_move in invoice_moves:
+                invoice.credit_control_notes =\
+                    invoice_move.credit_control_notes
+                break
+
+    @api.model
+    def _get_all_related_move_line(self):
```

Comment:
No need for this one according to other changes.
Replace it by:
```suggestion:+6
    def_get_payable_receivable_move_line(self):
        self.ensure_one()
        return self.move_id.line_ids.filtered(lambda l: l.account_id.internal_type' in' ['payable', 'receivable'])
```